src/ui/src/components/core/content/CoreText.vue

```diff
@@ -7,42 +7,38 @@
 		@click="handleClick"
 	>
 		<BaseEmptiness v-if="isEmpty" :component-id="componentId" />
-		<template v-else>
+		<SharedControlBar v-else>
 			<BaseMarkdown
 				v-if="fields.useMarkdown.value"
 				:raw-text="fields.text.value"
 				:style="contentStyle"
-			>
-			</BaseMarkdown>
-			<div v-else class="text-container">
-				<p class="plainText" :style="contentStyle">
-					{{ fields.text.value }}
-				</p>
-				<WdsButton
+			/>
+			<p v-else class="plainText" :style="contentStyle">
+				{{ fields.text.value }}
+			</p>
+
+			<template #actions>
+				<WdsCopyClipboardButton
 					v-if="fields.quickCopy.value"
-					class="copy-button"
-					variant="neutral"
-					size="smallIcon"
-					:class="{ copied: isCopied }"
-					@click="handleCopy"
-				>
-					<WdsIcon :name="isCopied ? 'check' : 'clipboard'" />
-				</WdsButton>
-			</div>
-		</template>
+					label="Copy text"
+					:value="fields.text.value"
+				/>
+			</template>
+		</SharedControlBar>
 	</div>
 </template>
 
 <script lang="ts">
 import {
 	createBooleanField,
+	createEnableCopyButtonField,
 	cssClasses,
 	primaryTextColor,
 } from "@/renderer/sharedStyleFields";
 import { getClick } from "@/renderer/syntheticEvents";
 import { FieldCategory, FieldControl, FieldType } from "@/writerTypes";
-import WdsButton from "@/wds/WdsButton.vue";
-import WdsIcon from "@/wds/WdsIcon.vue";
+import WdsCopyClipboardButton from "@/wds/WdsCopyClipboardButton.vue";
```

Comment:
Since this component imports floating UI, it's better to make it lazy to split the bundle.

```suggestion
import { defineAsyncComponentWithLoader } from "@/utils/defineAsyncComponentWithLoader";

const WdsCopyClipboardButton = defineAsyncComponentWithLoader({
	loader: () => import("@/wds/WdsCopyClipboardButton.vue"),
});
```